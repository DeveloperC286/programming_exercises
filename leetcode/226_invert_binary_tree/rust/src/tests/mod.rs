use super::*;

use insta::assert_debug_snapshot;

#[test]
fn test_single_node_with_left_and_right_tree() {
    // Given
    let tree = Some(Rc::new(RefCell::new(TreeNode {
        val: 2,
        left: Some(Rc::new(RefCell::new(TreeNode {
            val: 1,
            left: None,
            right: None,
        }))),
        right: Some(Rc::new(RefCell::new(TreeNode {
            val: 3,
            left: None,
            right: None,
        }))),
    })));

    // When
    let actual = invert_tree(tree);

    // Then
    assert_debug_snapshot!("test_single_node_with_left_and_right_tree", actual);
}

#[test]
fn test_multiple_nodes_with_left_and_right_tree() {
    // Given
    let tree = Some(Rc::new(RefCell::new(TreeNode {
        val: 4,
        left: Some(Rc::new(RefCell::new(TreeNode {
            val: 2,
            left: Some(Rc::new(RefCell::new(TreeNode {
                val: 1,
                left: None,
                right: None,
            }))),
            right: Some(Rc::new(RefCell::new(TreeNode {
                val: 3,
                left: None,
                right: None,
            }))),
        }))),
        right: Some(Rc::new(RefCell::new(TreeNode {
            val: 7,
            left: Some(Rc::new(RefCell::new(TreeNode {
                val: 6,
                left: None,
                right: None,
            }))),
            right: Some(Rc::new(RefCell::new(TreeNode {
                val: 9,
                left: None,
                right: None,
            }))),
        }))),
    })));

    // When
    let actual = invert_tree(tree);

    // Then
    assert_debug_snapshot!("test_multiple_nodes_with_left_and_right_tree", actual);
}
