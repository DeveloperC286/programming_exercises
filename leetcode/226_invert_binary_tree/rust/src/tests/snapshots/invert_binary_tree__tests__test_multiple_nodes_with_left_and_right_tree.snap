---
source: src/tests/mod.rs
expression: actual

---
Some(
    RefCell {
        value: TreeNode {
            val: 4,
            left: Some(
                RefCell {
                    value: TreeNode {
                        val: 7,
                        left: Some(
                            RefCell {
                                value: TreeNode {
                                    val: 9,
                                    left: None,
                                    right: None,
                                },
                            },
                        ),
                        right: Some(
                            RefCell {
                                value: TreeNode {
                                    val: 6,
                                    left: None,
                                    right: None,
                                },
                            },
                        ),
                    },
                },
            ),
            right: Some(
                RefCell {
                    value: TreeNode {
                        val: 2,
                        left: Some(
                            RefCell {
                                value: TreeNode {
                                    val: 3,
                                    left: None,
                                    right: None,
                                },
                            },
                        ),
                        right: Some(
                            RefCell {
                                value: TreeNode {
                                    val: 1,
                                    left: None,
                                    right: None,
                                },
                            },
                        ),
                    },
                },
            ),
        },
    },
)
